From a007d954f7a3c0b11f9c5bc77858415dc7e58439 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Mon, 10 Nov 2014 21:29:47 +0530
Subject: [PATCH 1/4] Owr: Use videoconvert instead of ercolorspace on Android

Not all Android devices have NEON support, and videconvert will do the right
thing using ORC
---
 local/owr_image_renderer.c     | 6 ++++++
 local/owr_local_media_source.c | 1 +
 local/owr_video_renderer.c     | 2 +-
 owr/owr.c                      | 4 ++--
 4 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/local/owr_image_renderer.c b/local/owr_image_renderer.c
index 0b9e5e0..2f5b9be 100644
--- a/local/owr_image_renderer.c
+++ b/local/owr_image_renderer.c
@@ -49,6 +49,12 @@
 #define DEFAULT_WIDTH 0
 #define DEFAULT_HEIGHT 0
 
+#if (defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR && !defined(__arm64__))
+#define VIDEO_CONVERT "videoconvert"
+#else
+#define VIDEO_CONVERT "videoconvert"
+#endif
+
 #define DEFAULT_MAX_FRAMERATE 0.0
 
 #define LIMITED_WIDTH 640
diff --git a/local/owr_local_media_source.c b/local/owr_local_media_source.c
index ef05128..f818483 100644
--- a/local/owr_local_media_source.c
+++ b/local/owr_local_media_source.c
@@ -59,6 +59,7 @@
 #elif defined(__ANDROID__)
 #define AUDIO_SRC "openslessrc"
 #define VIDEO_SRC "androidvideosource"
+#define VIDEO_CONVERT "videoconvert"
 
 #elif defined(__linux__)
 #define AUDIO_SRC  "pulsesrc"
diff --git a/local/owr_video_renderer.c b/local/owr_video_renderer.c
index ac2f80b..1478c74 100644
--- a/local/owr_video_renderer.c
+++ b/local/owr_video_renderer.c
@@ -50,7 +50,7 @@
 
 #define VIDEO_SINK "glimagesink"
 
-#if defined(__ANDROID__) || (defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR && !defined(__arm64__))
+#if (defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR && !defined(__arm64__))
 #define VIDEO_CONVERT "ercolorspace"
 #else
 #define VIDEO_CONVERT "videoconvert"
diff --git a/owr/owr.c b/owr/owr.c
index 267310c..2390c9e 100644
--- a/owr/owr.c
+++ b/owr/owr.c
@@ -107,7 +107,7 @@ GST_PLUGIN_STATIC_DECLARE(pulseaudio);
 GST_PLUGIN_STATIC_DECLARE(video4linux2);
 #endif
 
-#if defined(__ANDROID__) || (defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR && !defined(__arm64__))
+#if (defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR && !defined(__arm64__))
 GST_PLUGIN_STATIC_DECLARE(ercolorspace);
 #endif
 #endif
@@ -235,7 +235,7 @@ void owr_init()
     GST_PLUGIN_STATIC_REGISTER(pulseaudio);
     GST_PLUGIN_STATIC_REGISTER(video4linux2);
 #endif
-#if defined(__ANDROID__) || (defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR && !defined(__arm64__))
+#if (defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_IPHONE_SIMULATOR && !defined(__arm64__))
     GST_PLUGIN_STATIC_REGISTER(ercolorspace);
 #endif
 
-- 
2.2.2

